### ISSUE TITLE
ハイパーパラメータ自動チューニングの実装

### ISSUE THREAD

**Issue by synx-kawai** [synx-kawai/wavira#58]
## 概要
Optuna等を使用してハイパーパラメータの自動探索を行い、最適な学習設定を特定する。

## 背景
現在のハイパーパラメータは経験則に基づく固定値。体系的な探索で精度向上が期待できる。

## タスク
- [ ] Optunaの導入
  - [ ] 依存関係の追加 (requirements.txt)
  - [ ] 基本的なStudy設定
- [ ] 探索対象パラメータの定義
  - [ ] 学習率: [1e-5, 1e-2] (log scale)
  - [ ] バッチサイズ: [8, 16, 32, 64]
  - [ ] hidden_dim: [64, 128, 256]
  - [ ] num_layers: [1, 2, 3]
  - [ ] dropout: [0.0, 0.1, 0.2, 0.3, 0.5]
  - [ ] weight_decay: [0, 1e-5, 1e-4, 1e-3]
- [ ] 探索戦略
  - [ ] TPE (Tree-structured Parzen Estimator)
  - [ ] Pruning (Median Pruner)
  - [ ] 並列探索設定
- [ ] 探索スクリプトの実装
  - [ ] `scripts/tune_hyperparams.py`
  - [ ] 結果の保存と可視化

## コマンド例
```bash
python scripts/tune_hyperparams.py --n_trials 100 --data_dir data/crowd/
```

## 成果物
- [ ] ハイパーパラメータチューニングスクリプト
- [ ] 最適パラメータセット
- [ ] 探索結果の可視化（importance plot, optimization history）

## 関連
- Phase 4: 学習プロセスの最適化

**Comment by synx-kawai** [comment]
## 完了報告

### 実装内容

#### 新規作成ファイル
- `wavira/training/hyperparameter_tuning.py` - ハイパーパラメータチューニングモジュール
- `scripts/tune_hyperparams.py` - チューニングスクリプト
- `tests/test_hyperparameter_tuning.py` - テストコード（7テスト + 4 Optunaテスト）

### 実装機能

#### HyperparameterTuner（Optuna使用）
- TPE (Tree-structured Parzen Estimator) サンプラー
- Median Pruner による早期打ち切り
- 柔軟なSearchSpace定義
- 結果保存とパラメータ重要度分析

#### SimpleGridSearch（Optunaなし）
- Optuna不要の代替手法
- 組み合わせ総当たり探索
- 並列化なしのシンプルな実装

#### 探索対象パラメータ
- 学習率: [1e-5, 1e-2] (log scale)
- バッチサイズ: [8, 16, 32, 64]
- hidden_dim: [64, 128, 256]
- num_layers: [1, 2, 3]
- encoder_type: [transformer, lstm, bilstm]
- dropout: [0.0, 0.1, 0.2, 0.3, 0.5]
- weight_decay: [0, 1e-5, 1e-4, 1e-3]

### 使用方法

```bash
# Optunaを使用（要インストール: pip install optuna）
python scripts/tune_hyperparams.py --n_trials 100

# Grid searchを使用（Optunaなし）
python scripts/tune_hyperparams.py --use-grid-search
```

### 動作確認
- SimpleGridSearchテスト全パス
- Grid searchスクリプト動作確認済み
